internal/command/webbrowser/exec.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package webbrowser
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+// NewExecLauncher creates and returns a Launcher that just attempts to run
+// the executable at the given path, with the given URL as its first and
+// only argument.
+//
+// The given path must be ready to use, without reference to the PATH
+// environment variable. The caller can use [exec.LookPath] to prepare
+// a suitable path if searching PATH is appropriate.
+//
+// This is intended to allow overriding which browser to use using the
+// BROWSER environment variable on Unix-like systems, but the rules for
+// that are in "package main". [ParseBrowserEnv] implements parsing of the
+// value of that environment variable when the main package decides it's
+// appropriate to do so.
+func NewExecLauncher(execPath string) Launcher {
+	return execLauncher{
+		execPath: execPath,
+	}
+}
+
+type execLauncher struct {
+	execPath string
+}
+
+func (l execLauncher) OpenURL(url string) error {
+	cmd := &exec.Cmd{
+		Path: l.execPath,
+		Args: []string{l.execPath, url},
```

Comment:
q: Since the `l.execPath` is mentioned in the Path, do we need to mention it in the Args too?
a: Yes

Context: https://pkg.go.dev/os/exec#Cmd
> 	// Args holds command line arguments, including the command as Args[0].
>	// If the Args field is empty or nil, Run uses {Path}.
>	//
>	// In typical use, both Path and Args are set by calling Command.

Reply:
Indeed, this Go API is following the convention from [`execve`](https://www.man7.org/linux/man-pages/man2/execve.2.html) where the file to execute is conventionally specified in both places:

> `argv` is an array of pointers to strings passed to the new program as its command-line arguments.  By convention, the first of these strings (i.e., `argv[0]`) should contain the filename associated with the file being executed. [...]

It's a weird historical oddity, but it's required to give the child process the argument vector it's expecting.